x-pack/filebeat/input/cel/input.go

```diff
@@ -758,6 +760,25 @@ func newClient(ctx context.Context, cfg config, log *logp.Logger, reg *monitorin
 		maxSize := cfg.Resource.Tracer.MaxSize * 1e6
 		trace = httplog.NewLoggingRoundTripper(c.Transport, traceLogger, max(0, maxSize-margin), log)
 		c.Transport = trace
+	} else if cfg.Resource.Tracer != nil {
+		// We have a trace log name, but we are not enabled,
+		// so remove all trace logs we own.
+		ext := filepath.Ext(cfg.Resource.Tracer.Filename)
+		base := strings.TrimSuffix(cfg.Resource.Tracer.Filename, ext)
+		err = os.Remove(cfg.Resource.Tracer.Filename)
+		if err != nil && !errors.Is(err, fs.ErrNotExist) {
+			log.Errorw("failed to remove request trace log", "path", cfg.Resource.Tracer.Filename, "error", err)
+		}
+		paths, err := filepath.Glob(base + "-*" + ext)
+		if err != nil {
+			log.Errorw("failed to collect request trace log path names", "error", err)
+		}
+		for _, p := range paths {
+			err = os.Remove(p)
+			if err != nil && !errors.Is(err, fs.ErrNotExist) {
+				log.Errorw("failed to remove request trace log", "path", p, "error", err)
+			}
+		}
```

Comment:
The `ext :=` and `base :=` lines could come down here to where they're used, but more importantly...

The globbing here is looking for bad filenames. For example, with
```
resource.tracer.filename: logs/http-request-trace-*.ndjson
```
```
env.ID == "test_id:tracer_filename_sanitization_enabled"
```
the file name with sanitized ID is:
```
logs/http-request-trace-test_id_tracer_filename_sanitization_enabled.ndjson
```
and the clean-up globbing tries to delete things that wouldn't exist:
```
logs/http-request-trace-test_id_tracer_filename_sanitization_enabled-*.ndjson
```

Another issue is that the original file name config may have had `*` anywhere in the name, and may have had several or none of them. This is an unlikely edge case, but
```
resource.tracer.enabled: false
resource.tracer.filename: mylog
```
would delete `mylog-important-analysis-and-notes.txt`.

I was thinking that a solution would be to pull down the santization down into this function, so the original configuration value is available. However, there's a difference between the file name pattern and globbing. For example, `resource.tracer.filename: **` would use the file name `envidenvid`, but if used for clean-up globbing it would delete everything. I'm not sure what to do here.

Reply:
Moved the decls.

In the case that lumberjack is given a filename, which is always the case with our use, it will produce two kinds of paths, say given a name that resolves to "filename-id.ext", 1. `filename-id.ext` for the head of the logs and 2. [`filename-id-<RCF3339ish>.ext`](https://github.com/natefinch/lumberjack/blob/4cb27fcfbb0f35cb48c542c5ea80b7c1d18933d0/lumberjack.go#L257-L258) for the backups. The head is removed in the change here at [L768](https://github.com/efd6/beats/blob/8952f2ca59b490e49461f2b288d7e04271fb8c73/x-pack/filebeat/input/cel/input.go#L768) and then we work on the — potentially empty — list of backups that are identified by the glob. If they are not present, they don't need to be removed.

The filenames may not have invalid characters in their string since this has already been cleaned up by calling [`sanitizeFileName`](https://github.com/efd6/beats/blob/8952f2ca59b490e49461f2b288d7e04271fb8c73/x-pack/filebeat/input/cel/input.go#L126-L133) and [running `strings.ReplaceAll` on the `"*"` with the `env.ID`](https://github.com/efd6/beats/blob/8952f2ca59b490e49461f2b288d7e04271fb8c73/x-pack/filebeat/input/cel/input.go#L144-L147). These mutations to the filename happen before the `newClient` is called and the check for tracing happens and this removal is performed.

I will tighten the glob to match the RFC3339ish

Reply:
Yeah, sorry I made a mistake here. I thought the `-*` suffix was to match other env ID and thought the user might not match the pattern set by the user. On a morning walk I realized that was for the backups.

So it's just going for the named file, possibly with the current env ID, which I think its good. Tightening the glob is even better.

Side note: not changed here, but the `sanitizeFileName` name is a bit misleading since we only apply that function to the env ID, not the full file name.